### ISSUE TITLE
Electrical power balance

### ISSUE THREAD

**Issue by mugling** [CleverRaven/Cataclysm-DDA#17037]
Our current implementation of electrical power has two significant flaws. Firstly we do not model current so it's possible to start a `v12_combustion` by adding a tiny stack of `battery` to a `battery_motorbike`. Secondly vehicle batteries contain too much charge so you can boil many gallons of water using relatively small batteries.

I'd like to propose implementing CCA (cold cranking amps) which should provide a logical division between the SLI (starting) batteries and the storage variants. Each engine should require a given number of CCA to start, exacerbated by damage or faults (especially diesels in gold weather without glow plugs).

Secondly I'd like to propose rescaling the units to match real world value:
- Boiling 1L of water from ~20C requires aproximately 334kJ.
- An 80Ah battery at 12V can be expected to supply 1.2Kwh
- This is 4.3MJ and sufficient to boil just under 13L
- `hotplate` requires 12 charges of `battery` to boil 1L of `water_clean`
- `battery_truck` should therefore have 156 charges (27kJ/charge)
- Instead it has 8000 charges (aprox 50 times too much)

We obviously need some granularity, so if we went with 10kJ/charge:

| type | CCA | Ah | old charges | kJ @12v | new charges |
| --- | --- | --- | --- | --- | --- |
| storage | 50 | 400 | 40000 | 17280 | 2000 |
| motorbike | 100 | 40 | 2000 | 1728 | 200 |
| car | 250 | 80 | 4500 | 3456 | 350 |
| truck | 500 | 100 | 8000 | 4320 | 450 |
- At 334kJ/L a single charge of `water_clean` requires ~60kJ
- This would be 6 charges of `battery` using the new system
- Our crafting recipes therefore remain fairly close approximations
- `battery` however needs a lower `count` and `stack_size`, possibly 10

CCA should degrade with charge to prevent exploits such as filling with a single unit of `battery`:

![cca](https://cloud.githubusercontent.com/assets/12238606/15794273/82b3a824-29e1-11e6-835c-33ab878ad0ac.png)

Red, blue and green are motorbike, car and truck battery respectively. 100 CCA should be sufficient to start an engine with 1L displacement.

**tl;dr**
- Vehicle battery `capacity` reduced ~1:20
- `battery` stack size and count reduced ~1:10
- Crafting recipes remain similar
- Batteries provide `CCA`, each unit of which can start an equal number of units `displacement`

**Comment by Coolthulhu** [comment]
> battery however needs a lower count and stack_size, possibly 10

This would require a ton of changes and hurt granularity, which, in case of batteries, is really important.
Consider those:
* Flashlight power drain
* MP3 power drain
* UPS

It would be a lot of changing too.

> I'd like to propose implementing CCA (cold cranking amps) which should provide a logical division between the SLI (starting) batteries and the storage variants.

Slapping a truck battery on a car is trivial, so the requirement to have at least one truck battery on a vehicle with a big engine would result in complexity, but no depth.

> CCA should degrade with charge to prevent exploits such as filling with a single unit of battery

Now that batteries are removable by hand, we could just get rid of the whole battery refill thing. It never made any sense.

> Boiling 1L of water from ~20C requires aproximately 334kJ.

I don't think clean water crafting is supposed to reflect reality all that well. If we want realism here, it would probably be better to adjust crafting recipes that use water.

**Comment by mugling** [comment]
> hurt granularity,

We can scale for whatever granularity is required. The point I'm trying to make is that many tasks are way to cheap - in the case of boiling water by nearly 20x times. After the first electronics shop (and often before that) you're guaranteed never to run out.

> requirement to have at least one truck battery on a vehicle with a big engine would result in complexity, but no depth

You have to both charge it (using another vehicle initially) and then keep it charged. This makes smaller engines a more viable option and should hopefully limit the v12 wheelchair a little.

> Now that batteries are removable by hand, we could just get rid of the whole battery refill thing. It never made any sense.

Sounds good. A craftable jump-start pack (using a sensible amount of charge) would be a suitable replacement.

> If we want realism here, it would probably be better to adjust crafting recipes that use water.

Yes, boiling water is a particularly egregious example.

**Comment by Coolthulhu** [comment]
> We can scale for whatever granularity is required. The point I'm trying to make is that many tasks are way to cheap - in the case of boiling water by nearly 20x times. After the first electronics shop (and often before that) you're guaranteed never to run out.

Good point, but that's mostly because recipes are just so cheap.
It gets even worse when you get the lamp oil cooker and hotplate recipes become pretty much free.

> You have to both charge it (using another vehicle initially) and then keep it charged.

Slap it onto a car that is broken because no wheels, but still has battery, engine, controls and alternator. Start it up then remove the smaller battery. Charge new battery up. Move to new car and have an alternator on it. Problem permanently solved.

> This makes smaller engines a more viable option and should hopefully limit the v12 wheelchair a little.

It could make them necessary to start, then they would fall into meaninglessness just as fast as before, except as a backup due to bad luck somehow making battery charge get weird.

A smart battery system should charge best (highest CCA) batteries first and drain the worst first. With said smart battery system, it would be just a matter of keeping a truck battery on your vehicle.

Without it, there would be weird metagame where you'd want to keep a charged truck battery just to start vehicles, then would remove it and have the vehicle run on other batteries. It would be incredibly annoying, not realistic and wouldn't add anything good to the game.

V12 wheelchair would be better solved by removing the ability to power wheelchair wheels with an engine and limiting amount of stuff that can be mounted per tile.

**Comment by mugling** [comment]
> With said smart battery system...

Charging highest CCA first would indeed be a requirement. In a similar manner charge should be drained from the lowest CCA for non-starting tasks.

> ...it would be just a matter of keeping a truck battery on your vehicle.

You'd actually need a pair of truck batteries for some of the larger engines - possibly more in adverse conditions. This increases the weight and particularly size of your vehicle. It makes V12 motorbike less appealing.

Battery power is roughly 20x too plentiful. There's a number of ways of rescaling of course.

**Comment by Coolthulhu** [comment]
> This increases the weight and particularly size of your vehicle. It makes V12 motorbike less appealing.

Which doesn't mean much at all. V12 has enough power to make 2 truck batteries negligible. At the moment, V12 weighs as much as 10 truck batteries.

Without a good redesign of the acceleration, speed and fuel consumption formulas, it will all boil down to V12 and batteries for it.
Without a good system for limiting vehicle part volume per tile, the only thing that would change here would be V12 wheelchairs splitting into V12 deathbikes and V8 wheelchairs.

> Battery power is roughly 20x too plentiful.

For crafting recipes. It's about right for electric engines and floodlights.

**Comment by Zireael07** [comment]
Nice graph, @mugling!

**Comment by pingpong2011** [comment]
To comment, I honestly like having excess power at higher tools. Makes the game less about maintenance and more about getting out and doing stuff. Traditionally cars were "overpowered" in that they can withstand good amounts of damage, they can be used for near-everything, and they have ridiculous storage capacity. It wouldn't be unreasonable to increase the cost of making clean water and removing the ability to refill a battery with batteries, but to make them "realistic" is such a pain. If that happened, I'd want an option for a steam powered car, because water and charcoal is plentiful.

**Comment by mugling** [comment]
> If that happened, I'd want an option for a steam powered car, because water and charcoal is plentiful.

That could be interesting it it's own right

**Comment by mugling** [comment]
Lead-acid batteries have reasonably consistent energy densities:

* **specific energy** 33-42Wh/kg
* **energy density** 60-110L/kg

Some empiric data from averaging various IRL batteries:

| part      | weight | vol (L)  | Ah  | Wh   | kJ   | Wh/kg | Wh/L | acid (L) |
|-----------|--------|------|-----|------|------|-------|------|----------|
| motorbike | 5.3    | 2.0  | 15  | 180  | 648  | 34    | 90   | 0.8      |
| car       | 18.9   | 6.2  | 60  | 720  | 2592 | 38    | 116  | 2.4      |
| truck*     | 21.0   | 9.8  | 70  | 840  | 3024 | 40    | 86   | 2.8      |
| storage   | 31.4   | 12.5 | 110 | 1320 | 4752 | 42    | 106  | 3.8      |

`*` usually a pair of 12V batteries so for our purposes this category is represented by a high performance commercial battery providing high cold-cranking amps (CCA)

Formatted nicely as in-game values:

| part      | weight | vol | (Ah)  | charges   | acid |
|-----------|--------|-----|-----|------|----------|
| motorbike | 6.1    | 8   | 17  | 750  | 3        |
| car       | 18.3   | 25  | 58  | 2500 | 10       
| truck     | 20.8   | 40  | 69  | 3000 | 12       |
| storage   | 33.1   | 52  | 116 | 5000 | 15       |

Accounting for @Coolthulhu's earlier concerns about granularity this now uses `1 charge = 1kJ`

When combined with #17166 and horsepower data for engines this should produce a sanely balanced system for vehicle power with suitable reference units for anyone wishing to add new units.

**Comment by Coolthulhu** [comment]
Keep in mind that current "storage batteries" are things that are mounted on electric vehicles and are supposed to make them run long enough for them to actually not suck.

Averaging IRL values isn't the best solution here - look at how similar car and truck batteries are now. Some of the IRL trucks got a car battery in the game (flatbed truck). It's better to give them a car battery than to have 2 kinds of batteries with nearly identical stats.

What you named "storage battery" would be better off being the truck battery.
Meanwhile we also need an electric car battery, which would be way bigger than all the batteries in the table combined.

Linear charges/weight ratio isn't good. It makes the battery size irrelevant, as in this case you can (and should) just stack more of them.

**Comment by pingpong2011** [comment]
I was thinking, to fix "filling" a battery, one could have an emergency battery starter. I forgot that I've literally had a Ni-Cad, cigarette lighter plug one in my home. I wouldn't trust how well it'd work in real life,  but the tool could use 50-100 batteries each attempt to start a car (maybe with a warning if the engine is faulty).

**Comment by mugling** [comment]
> Keep in mind that current "storage batteries" are things that are mounted on electric vehicles and are supposed to make them run long enough for them to actually not suck.

We can arbitrarily scale up in proportion as necessary

| part      | weight | vol | (Ah)  | charges   | acid |
|-----------|--------|-----|-----|------|----------|
| before   | 31.7   | 50  | 111 | 4800 | 15       |
| after     | 63.4   | 100  | 222  | 5600 | 30       |

> look at how similar car and truck batteries are now... It's better to give them a car battery than to have 2 kinds of batteries with nearly identical stats.

Agreed, especially as there isn't really a discrete truck battery IRL - they're almost always pairs of standard SLI batteries.

> Meanwhile we also need an electric car battery, which would be way bigger than all the batteries in the table combined.

That could either be multiple storage batteries installed in parallel or more appropriately lithium ion or an entirely new battery technology?

> as in this case you can (and should) just stack more of them.

Sensible limits on stacking will be applied one #17062 is a thing

**Comment by Coolthulhu** [comment]
> truck battery could be replaced by performance battery with higher CCA

CCA mechanics will not be a good thing until vehicle weight is penalized heavily. They're not intuitive, possibly very tedious and I don't see them adding much depth to the game (complexity is another matter).
They will need a really good reason to exist before they exist.

> That could either be multiple storage batteries installed in parallel or more appropriately lithium ion or an entirely new battery technology?

Could be either (or both).

Also, small and medium storage batteries exist. They will need to be handled here somehow, while keeping in mind that small storage battery is uncraftable, doesn't spawn "in the wild", is required for multiple recipes and is currently obtained by disassembling big storage batteries.
The whole "modular battery" idea is not exactly a success and we'd be better off without it, but many recipes rely on it as the source of small storage batteries.

**Comment by mugling** [comment]
> CCA mechanics will not be a good thing until vehicle weight is penalized heavily ... They will need a really good reason to exist before they exist.

The only real difference between SLI and storage batteries is their CCA/discharge properties so if we don't intend to model that then we don't really need separate lead-acid storage batteries.

> The whole "modular battery" idea is not exactly a success and we'd be better off without it, but many recipes rely on it as the source of small storage batteries.

I've not really explored this further and I'm not sure when it was added but why do you feel it to be an inherently bad idea? I can see a potential revision involving uncraftable lithium-ion cells being joined together to create larger units?

**Comment by Coolthulhu** [comment]
> I've not really explored this further and I'm not sure when it was added but why do you feel it to be an inherently bad idea?

There is no need for medium storage batteries. There are small batteries for use as components, there are large batteries for electric vehicles and stationary storage and there are medium batteries that only exist as an intermediate stage between those two and have no role in the game.

Same problem is with other vehicle batteries, where you don't really need medium batteries for anything. Though there aren't really any medium batteries - there are small ones (up to truck) and large ones (storage).
Small ones are for combustion engines. Large ones are for electric engines. CCA wouldn't help here since installing a truck battery isn't a problem.

Modular batteries could exist for realism, but they don't have a good gameplay role.

> I can see a potential revision involving uncraftable lithium-ion cells being joined together to create larger units?

Both "ends" of the recipe should have their roles. Joining 8 small batteries to create a medium battery will not change anything. Joining 100 small batteries to create a large battery would, but then we'd have a recipe that takes/creates 100 objects.

**Comment by mugling** [comment]
How about if lithium technology was 5x as energy dense:

| item            | vol | stack | weight | charges |
|-----------------|-----|-------|--------|---------|
| lithium cell    | 1   | 5     | 0.1    | 100     |
| lithium battery | 1   |       | 0.5    | 500     |
| storage battery | 20   |       | 10      | 10'000    |

**Comment by kevingranade** [comment]
Super old suggestion thread, closing for inactivity.
